### ISSUE TITLE
Testnet is no longer syncing past block 299187

### ISSUE THREAD

**Issue by oliverw** [zcash/zcash#3568]
### Describe the issue
Testnet is no longer syncing past block 299187.

### Can you reliably reproduce the issue?

`zcashd -server -testnet -rpcuser=user -rpcpassword=pass -rpcbind=0.0.0.0 -rpcallowip=::/0 -port=8233 -rpcport=8232 -addnode -addnode=betatestnet.z.cash -zmqpubrawtx=tcp://0.0.0.0:3000 -zmqpubrawblock=tcp://0.0.0.0:3000 -zmqpubhashtx=tcp://0.0.0.0:3000 -zmqpubhashblock=tcp://0.0.0.0:3000
`
### Expected behaviour
Syncing should complete.

### Actual behaviour + errors
Syncing never completes. Stays forever at:

> Downloading blocks | 299187 / ~302100 (99%)
> Connections | 8
> Network solution rate | 2542 Sol/s
> 

### The version of Zcash you were using:
Run `zcashd --version` to find out

Zcash Daemon version v2.0.0

### Machine specs:
- OS: Ubuntu 16.04

**Comment by rex4539** [comment]
Probably duplicate of https://github.com/zcash/zcash/issues/3552

Fixed in master already.

**Comment by oliverw** [comment]
A fresh build from master does not sync either past block 299187.

**Comment by guybrush** [comment]
same here

```
$ git rev-parse HEAD
72f6925de1c78b8481d79a71a5d94d4f3324a594
$ zcash-cli getinfo
{
  "version": 2000050,
  "protocolversion": 170007,
  "walletversion": 60000,
  "balance": 0.00000000,
  "blocks": 299187,
  "timeoffset": 0,
  "connections": 7,
  "proxy": "",
  "difficulty": 193722.9601781637,
  "testnet": true,
  "keypoololdest": 1538561052,
  "keypoolsize": 101,
  "paytxfee": 0.00000000,
  "relayfee": 0.00000100,
  "errors": ""
}
$ zcash-cli getblocktemplate
error code: -10
error message:
Zcash is downloading blocks...
```

**Comment by daira** [comment]
This may be because there are no blocks beyond 299187 on the *public* testnet branch created by the fork, yet. (We have tested privately that it's possible to mine blocks after that, but you'd have to connect to the nodes used in that testing to see those blocks.)

**Comment by oliverw** [comment]
But it definitely is.

**Comment by bitcartel** [comment]
"downloading blocks" is a "maxtipage" issue - see discussion here: https://github.com/zcash/zcash/pull/3559#pullrequestreview-161399395

**Comment by daira** [comment]
Here's what's going on: there are currently no blocks on the public testnet after 299187. But the estimate of how many blocks there "should" be is based on the current time. That's why a newly started node (running the code currently on master) will think that downloading has not completed. If a node has been running since before the stall, it won't see this problem because the "finished downloading" state will have latched. This problem will resolve itself, for 2.0.1-rc1 and 2.0.1 nodes, once mining on the public testnet starts up again. Only the miner of block 299188 will need to use `-maxtipage`.

**Comment by oliverw** [comment]
@daira Thanks for the thorough explanation Daira.

**Comment by ruguo220088** [comment]
> Here's what's going on: there are currently no blocks on the public testnet after 299187. But the estimate of how many blocks there "should" be is based on the current time. That's why a newly started node (running the code currently on master) will think that downloading has not completed. If a node has been running since before the stall, it won't see this problem because the "finished downloading" state will have latched. This problem will resolve itself, for 2.0.1-rc1 and later node versions, once mining on the public testnet starts up again. Only the miner of block 299188 will need to use `-maxtipage`.

I use -maxtipage, but it can not work also. the block height is 299187, can not downloading the next blocks.  help, please.

**Comment by oliverw** [comment]
@ruguo220088 Did you even read Daira's response?

**Comment by daira** [comment]
@oliverw: please maintain a constructive tone, and bear in mind that English may not be some users' first language.

@ruguo220088: it may be better for you to wait a little while until mining has resumed on testnet.

**Comment by ruguo220088** [comment]
> @oliverw: please maintain a constructive tone, and bear in mind that English may not be some users' first language.
> 
> @ruguo220088: it may be better for you to wait a little while until mining has resumed on testnet.

What I don't understand is that since there is no new block after 299187, why does the explorer.testnet show that the current block is already 303307, and the currency I applied for zec is also at 303014.

**Comment by daira** [comment]
You're right, mining on testnet has already resumed. Are you sure that you're running version 2.0.1-rc1?

**Comment by oliverw** [comment]
@daira No offense meant actually.

**Comment by ruguo220088** [comment]
> You're right, mining on testnet has already resumed. Are you sure that you're running version 2.0.1-rc1?

it works now, thanks a lot.
